complement/pipeline.yml

```diff
@@ -37,3 +40,63 @@ steps:
       automatic:
         - exit_status: 128
           limit: 3
+
+  - command:
+      # Build the Synapse for Complement docker image, which is located at:
+      # https://github.com/matrix-org/complement/blob/master/dockerfiles/Synapse.Dockerfile.
+      - docker build -t complement-synapse -f dockerfiles/Synapse.Dockerfile dockerfiles/
+      # Run the tests!
+      - COMPLEMENT_BASE_IMAGE=complement-synapse go test -v -tags "synapse_blacklist" ./tests
+    label: "\U0001F9EA Complement / Synapse Monolith / :go: 1.15"
+    agents:
+      # As we provide access to the host docker socket below, use an ephemeral agent which
+      # will be destroyed after this job is complete. This will prevent any potentially
+      # malicious code running in this build agent from tampering with any further jobs.
+      queue: "medium-ephemeral"
```

Comment:
shower thought on this: even the `ephemeral` build agents still have the buildkite agent token, which gives you the ability to run arbitrary jobs on *other* agents, so this isn't going to cut it. I think we'll have to lock the build down to trusted branches, which means you may as well go back to `medium`, since they are faster.

sorry for the runaround.

Reply:
Ah, darn OK. Have removed them then. Be sure to turn off Contributor builds as well.

Reply:
we should probably stick a warning in at the top of the file to remind ourselves not to enable contrib PRs, and why:

```
# WARNING: do not enable builds of third-party PRs on this pipeline. The pipeline is configured to
# give access to the Docker socket to the Complement test binaries, so a malicious fork of
# Complement could start a Docker container which extracts the Buildkite agent token and hence
# compromises this and other Buildkite agents.
```

Reply:
Looking at https://buildkite.com/docs/pipelines/environment-variables#environment-variable-precedence can't we just clobber the agent token in this pipeline by setting some bogus value?

Reply:
for this pipeline, sure, but we still have the problem that the token exists somewhere on the parent host, and we are giving unrestricted access to the parent host via the docker socket.

Reply:
Have added the warning.